app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.test.js

```diff
@@ -1,30 +1,113 @@
+  const props = {school: testHighSchool};
+  const el = testEl(props);
+  const dash = mount(elWrappedInContext(el, context));
+  expect(dash.find('SelectHouse').length > 0).toEqual(true);
+});
+
+it('does not display house for Healey', () => {
+  const context = testContext({districtKey: 'somerville'});
+  const props = {school: testSchool};
+  const el = testEl(props);
+  const dash = mount(elWrappedInContext(el, context));
+  expect(dash.find('SelectHouse').length > 0).toEqual(false);
+});
+
+it('displays counselor for SHS', () => {
+  const context = testContext({districtKey: 'somerville'});
+  const props = {school: testHighSchool};
+  const el = testEl(props);
+  const dash = mount(elWrappedInContext(el, context));
+  expect(dash.find('SelectCounselor').length > 0).toEqual(true);
+});
 
-  it('renders at least one bar chart', () => {
-    expect(dash.find('DashboardBarChart').length > 0).toEqual(true);
-  });
+it('does not display counselor for Healey', () => {
+  const context = testContext({districtKey: 'somerville'});
+  const props = {school: testSchool};
+  const el = testEl(props);
+  const dash = mount(elWrappedInContext(el, context));
+  expect(dash.find('SelectCounselor').length > 0).toEqual(false);
+});
 
-  it('renders a student list', () => {
-    expect(dash.find('StudentsTable').length).toEqual(1);
-  });
+it('displays grade for SHS', () => {
+  const context = testContext({districtKey: 'somerville'});
+  const props = {school: testHighSchool};
+  const el = testEl(props);
+  const dash = mount(elWrappedInContext(el, context));
+  expect(dash.find('SelectGrade').length > 0).toEqual(true);
+});
+
+it('displays grade for Healey', () => {
+  const context = testContext({districtKey: 'somerville'});
+  const props = {school: testSchool};
+  const el = testEl(props);
+  const dash = mount(elWrappedInContext(el, context));
+  expect(dash.find('SelectGrade').length > 0).toEqual(true);
+});
+
+it('renders at least one bar chart', () => {
+  const context = testContext({districtKey: 'somerville'});
+  const props = {school: testSchool};
+  const el = testEl(props);
+  const dash = mount(elWrappedInContext(el, context));
+  expect(dash.find('DashboardBarChart').length > 0).toEqual(true);
+});
+
+it('renders a student list', () => {
+  const context = testContext({districtKey: 'somerville'});
+  const props = {school: testSchool};
+  const el = testEl(props);
+  const dash = mount(elWrappedInContext(el, context));
+  expect(dash.find('StudentsTable').length > 0).toEqual(true);
+});
 
-  it('renders a date range selector', () => {
-    expect(dash.find('SelectTimeRange').length).toEqual(1);
-  });
+it('renders a date range selector', () => {
+  const context = testContext({districtKey: 'somerville'});
+  const props = {school: testSchool};
+  const el = testEl(props);
+  const dash = mount(elWrappedInContext(el, context));
+  expect(dash.find('SelectTimeRange').length > 0).toEqual(true);
```

Comment:
Given that this code now works differently per district, it would be good to add at least smoke tests for that.  Here's how this works in the absences page: https://github.com/studentinsights/studentinsights/blob/master/app/assets/javascripts/school_absences/SchoolAbsencesDashboard.test.js#L211

Reply:
@kevinrobinson I've got most of your suggestions implemented, but the CI build is hassling me about these snapshots. Is there something I should be sending in this branch to be able to write snapshots?

Reply:
@edavidsonsawyer Yeah!  So with snapshots, you need to update this locally and push up the changes.

Conceptually, snapshots are the same as any other test - your PR has to include how the expectations for the tests are different.  The only wrinkle is that Jest makes the snapshot expectations for you.  It should prompt you to do this when you run `yarn test` locally, something like this:

![image](https://user-images.githubusercontent.com/1056957/47658927-41196600-db6a-11e8-91a8-a09fa08b5d2d.png)

Here's a direct link to the Jest docs about this: https://jestjs.io/docs/en/snapshot-testing.html#updating-snapshots or if you our setup isn't working quite right or is confusing, let me know and I can take a look with you so we can make this work nicely and remove any friction.

Reply:
@kevinrobinson got it now, sorry that took so long. I forgot to merge master into this branch and it was messing up the snapshots here, but not locally.

Reply:
@edavidsonsawyer no worries!  Is this ready to change from WIP and to deploy now?  I can do that later today if this is ready to 🚢 👍

Reply:
@kevinrobinson sure is! thanks again for the review!

Reply:
no problemo!